File: lab2/gr8sub1/Павлючук_Юлия/src/fork_example.py
#!/usr/bin/env python3
import os
import sys
import time

def main():
    print(f"Parent start: PID={os.getpid()}", flush=True)

    children = []
    for i in range(2):  
        pid = os.fork()
        if pid == 0:
            # дочерний процесс
            print(f"child_{i}: PID={os.getpid()}, PPID={os.getppid()}", flush=True)
            time.sleep(10)
            sys.exit(10 + i)
        else:
            # родительский процесс: запоминаем PID ребёнка
            children.append(pid)

  
    for pid in children:
        finished_pid, status = os.waitpid(pid, 0)
        exit_code = os.WEXITSTATUS(status)
        print(f"Parent: child {finished_pid} exited with code {exit_code}", flush=True)

    print(f"Parent finish: PID={os.getpid()}", flush=True)

if __name__ == "__main__":
    main()

File: lab2/gr8sub1/ГУЗОВА_АННА/src/fork_example.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <sys/wait.h>

int main() {
    printf("[parent] start, PID=%d\n", getpid());
    fflush(stdout);

    pid_t children[2];

    for (int i = 0; i < 2; ++i) {
        pid_t pid = fork();
        if (pid < 0) {
            perror("fork");
            exit(EXIT_FAILURE);
        }
        if (pid == 0) {
            printf("[child %c] PID=%d, PPID=%d\n",
                   'A' + i, getpid(), getppid());
            fflush(stdout);
            _exit(EXIT_SUCCESS);
        }
        children[i] = pid;
    }

    for (int i = 0; i < 2; ++i) {
        int status;
        pid_t pid = waitpid(children[i], &status, 0);
        if (pid == -1) {
            perror("waitpid");
            exit(EXIT_FAILURE);
        }
        printf("[parent] child %d exited, status=%d\n", pid, WEXITSTATUS(status));
    }

    printf("[parent] done\n");
    return 0;
}
